cl/corpus_importer/tasks.py

```diff
@@ -1570,11 +1574,139 @@ def download_pacer_pdf_by_rd(
+
+    :param court_id: A CourtListener court ID to query the confirmation page.
+    :param pacer_doc_id: The pacer_doc_id to query the confirmation page.
+    :return: The PACER document number is available or an empty string if not.
+    """
+
+    recap_email_user = User.objects.get(username="recap-email")
+    cookies = get_or_cache_pacer_cookies(
+        recap_email_user.pk, settings.PACER_USERNAME, settings.PACER_PASSWORD
+    )
+    s = PacerSession(cookies=cookies)
+    doc_num_report = DownloadConfirmationPage(court_id, s)
+    doc_num_report.query(pacer_doc_id)
+    data = doc_num_report.data
+    return data.get("document_number", "")
+
+
+def get_document_number_appellate(
+    court_id: str,
+    pacer_doc_id: str,
+    pacer_case_id: str,
+    cookies: RequestsCookieJar,
+    magic_number: str,
+) -> tuple[str, Response | None, str | None]:
+    """A wrapper to get the PACER document number either from the download
+    confirmation page or from the PDF document.
+
+    :param court_id: A CourtListener court ID to query the confirmation page.
+    :param pacer_doc_id: The pacer_doc_id to query the confirmation page.
+    :param pacer_case_id: The pacer_case_id to query the free document.
+    :param cookies: The cookies of a logged in PACER session
+    :param magic_number: The magic number to fetch PACER documents for free.
+    :return: A tuple containing the PACER document number is available or an
+    empty string if not, a requests.Response object usually containing a PDF,
+    or None if that wasn't possible, and a string representing the error if
+    there was one.
+    """
+
+    # Try to download the free appellate PDF.
+    pdf_response, r_msg = download_free_appellate_pacer_pdf(
+        court_id,
+        pacer_doc_id,
+        pacer_case_id,
+        cookies,
+        magic_number,
+    )
+    document_number = ""
+    if court_id in ("ca8", "ca11", "cadc"):
+        # For ca8, ca11 and cadc the PACER document number is not available
+        # in the PDF, try to get it directly from the Download confirmation
+        # page.
+        document_number = get_document_number_from_confirmation_page(
+            court_id, pacer_doc_id
+        )
+    elif pdf_response:
+        pdf_bytes = pdf_response.content
+        # For other jurisdictions try first to get it from the PDF document.
+        dn_response = microservice(
+            service="document-number",
+            file_type="pdf",
+            file=pdf_bytes,
+        )
+        if dn_response.ok:
+            document_number = dn_response.text
+
+    # If we still don't have the document number fall back on the download
+    # confirmation page except for ca8, ca11, and cadc (we've already tried)
+    if not document_number and court_id not in ("ca8", "ca11", "cadc"):
+        document_number = get_document_number_from_confirmation_page(
+            court_id, pacer_doc_id
+        )
```

Comment:
Thanks, yeah, I've indented this block so now the `if` is simplified to `if not document_number:`
However, I also had to change a bit the if-else statements above, since we need to consider the following possibilities:

We can have or not a `pdf_response` depending if the PDF is available or not.

Even if we have a `pdf_response` that does not guarantee that we can get the document number from the PDF so we need to check that the doctor response is ok and the response text is not blank.

If after it we still don't have the document number we try to get it from the download confirmation page.